sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_custom_form_client.py

```diff
@@ -0,0 +1,296 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=protected-access
+
+from typing import (  # pylint: disable=unused-import
+    Union,
+    Optional,
+    Any,
+    IO,
+    Iterable,
+    TYPE_CHECKING,
+)
+import six
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.polling import LROPoller
+from azure.core.polling.base_polling import LROBasePolling  # pylint: disable=no-name-in-module,import-error
+from azure.core.exceptions import HttpResponseError
+from ._generated.models import AnalyzeOperationResult, Model
+from ._generated._form_recognizer_client import FormRecognizerClient as FormRecognizer
+from ._generated.models import TrainRequest, TrainSourceFilter
+from ._base_client import FormRecognizerClientBase
+from ._response_handlers import (
+    prepare_unlabeled_result,
+    prepare_labeled_result,
+)
+from ._helpers import get_pipeline_response, get_content_type, POLLING_INTERVAL
+from ._models import (
+    ModelInfo,
+    ModelsSummary,
+    CustomModel,
+    CustomLabeledModel,
+)
+if TYPE_CHECKING:
+    from ._credential import FormRecognizerApiKeyCredential
+
+
+class CustomFormClient(FormRecognizerClientBase):
+    """CustomFormClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of FormRecognizerApiKeyCredential if using an API key.
+    :type credential: ~azure.ai.formrecognizer.FormRecognizerApiKeyCredential
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):
+        # type: (str, FormRecognizerApiKeyCredential, Any) -> None
+        super(CustomFormClient, self).__init__(credential=credential, **kwargs)
+        self._client = FormRecognizer(
+            endpoint=endpoint, credential=credential, pipeline=self._pipeline
+        )
+
+    @distributed_trace
+    def begin_training(self, source, source_prefix_filter="", include_sub_folders=False, **kwargs):
```

Comment:
What happen if `source_prefix_filter=None`? Should this be the default? (open question not suggestion)

Reply:
It's optional, the service team specifically asked that I default it to `""`